File: EmbedCaption/EmbedCaption/frmMain.vb
Public Class frmMain

#Region "フォームイベント"

	''' <summary>
	''' フォームロード時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub frmMain_Load(sender As Object, e As EventArgs) Handles MyBase.Load

		Me.Text = My.Application.Info.ProductName & " Ver." &
			My.Application.Info.Version.ToString

		XmlSettings.LoadFromXmlFile()
		Me.txtCSVFile.Text = XmlSettings.Instance.CSVFile
		Me.txtParentFolder.Text = XmlSettings.Instance.ParentFolder
		Me.txtOutputFolder.Text = XmlSettings.Instance.OutputFolder
		Me.txtLogFolder.Text = XmlSettings.Instance.LogFolder
		Me.numYStart.Value = XmlSettings.Instance.YStart

	End Sub

	''' <summary>
	''' フォームクロージング
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub frmMain_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing

		If MessageBox.Show("アプリケーションを終了します" & vbNewLine & "よろしいですか？", "確認", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = DialogResult.No Then
			e.Cancel = True
		Else
			XmlSettings.LoadFromXmlFile()
			XmlSettings.Instance.CSVFile = Me.txtCSVFile.Text
			XmlSettings.Instance.ParentFolder = Me.txtParentFolder.Text
			XmlSettings.Instance.OutputFolder = Me.txtOutputFolder.Text
			XmlSettings.Instance.LogFolder = Me.txtLogFolder.Text
			XmlSettings.Instance.YStart = Me.numYStart.Value
			XmlSettings.SaveToXmlFile()
		End If

	End Sub

#End Region

#Region "オブジェクトイベント"

	''' <summary>
	''' テキストボックスドラッグエンター時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub txt_DragEnter(sender As Object, e As DragEventArgs) Handles txtCSVFile.DragEnter, txtParentFolder.DragEnter, txtOutputFolder.DragEnter, txtLogFolder.DragEnter

		If e.Data.GetDataPresent(DataFormats.FileDrop) Then
			e.Effect = DragDropEffects.Copy
		Else
			e.Effect = DragDropEffects.None
		End If

	End Sub

	''' <summary>
	''' CSVファイルドロップ時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub txtCSVFile_DragDrop(sender As Object, e As DragEventArgs) Handles txtCSVFile.DragDrop

		Dim strFiles As String() = CType(e.Data.GetData(DataFormats.FileDrop, False), String())
		If System.IO.File.Exists(strFiles(0)) Then
			Me.txtCSVFile.Text = strFiles(0)
		End If

	End Sub

	''' <summary>
	''' 親フォルダドロップ時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub txtParentFolder_DragDrop(sender As Object, e As DragEventArgs) Handles txtParentFolder.DragDrop

		Dim strFiles As String() = CType(e.Data.GetData(DataFormats.FileDrop, False), String())
		If System.IO.Directory.Exists(strFiles(0)) Then
			Me.txtParentFolder.Text = strFiles(0)
		End If

	End Sub

	''' <summary>
	''' 出力フォルダドロップ時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub txtOutputFolder_DragDrop(sender As Object, e As DragEventArgs) Handles txtOutputFolder.DragDrop

		Dim strFiles As String() = CType(e.Data.GetData(DataFormats.FileDrop, False), String())
		If System.IO.Directory.Exists(strFiles(0)) Then
			Me.txtOutputFolder.Text = strFiles(0)
		End If

	End Sub

	''' <summary>
	''' ログフォルダドロップ時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub txtLogFolder_DragDrop(sender As Object, e As DragEventArgs) Handles txtLogFolder.DragDrop

		Dim strFiles As String() = CType(e.Data.GetData(DataFormats.FileDrop, False), String())
		If System.IO.Directory.Exists(strFiles(0)) Then
			Me.txtLogFolder.Text = strFiles(0)
		End If

	End Sub

	''' <summary>
	''' 実行ボタン押下時
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Private Sub btnRun_Click(sender As Object, e As EventArgs) Handles btnRun.Click

		If Not System.IO.File.Exists(Me.txtCSVFile.Text) Then
			MessageBox.Show("CSVファイルが指定されていません", "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Exit Sub
		End If
		If Not System.IO.Directory.Exists(Me.txtParentFolder.Text) Then
			MessageBox.Show("親フォルダが指定されていません", "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Exit Sub
		End If
		If Not System.IO.Directory.Exists(Me.txtOutputFolder.Text) Then
			MessageBox.Show("出力フォルダが指定されていません", "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Exit Sub
		End If
		If Not System.IO.Directory.Exists(Me.txtLogFolder.Text) Then
			MessageBox.Show("ログ保存フォルダが指定されていません", "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Exit Sub
		End If

		If MessageBox.Show("キャプション挿入処理を開始します" & vbNewLine & "よろしいですか？", "確認", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = DialogResult.No Then
			Exit Sub
		End If

		EnableControls(Me, False)

		Dim strCSVFile As String = Me.txtCSVFile.Text
		Dim strParentFolder As String = Me.txtParentFolder.Text
		Dim strOutputFolder As String = Me.txtOutputFolder.Text
		Dim strLogFolder As String = Me.txtLogFolder.Text

		Try
			Me.lstResult.Items.Clear()
			WriteLstResult(Me.lstResult, "CSVファイル：" & System.IO.Path.GetFileName(strCSVFile))
			WriteLstResult(Me.lstResult, "親フォルダ：" & strParentFolder)
			WriteLstResult(Me.lstResult, "出力フォルダ：" & strOutputFolder)
			WriteLstResult(Me.lstResult, "ログフォルダ：" & strLogFolder)
			WriteLstResult(Me.lstResult, "Y軸起点：" & Me.numYStart.Value)

			Dim iRecCount As Integer = 0
			Using parser As New CSVParser(strCSVFile, System.Text.Encoding.GetEncoding("Shift_JIS"))
				parser.Delimiter = vbTab    'タブ区切り
				parser.HasFieldsEnclosedInQuotes = True
				parser.TrimWhiteSpace = False

				iRecCount = 0

				parser.ReadFields() 'ヘッダ業を読み飛ばす
				'最終行まで読み込み
				While Not parser.EndOfData
					iRecCount += 1
					Dim row As String() = parser.ReadFields()
					'子フォルダ以下の画像ファイルをすべて配列に列挙する
					Dim strFiles As String() = GetFilesMostDeep(strParentFolder & "\" & row(0), {"*.tif"})

				End While
			End Using

		Catch ex As Exception

			MessageBox.Show(ex.Message, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

		Finally

			EnableControls(Me, True)

		End Try
	End Sub

#End Region

End Class

File: CreateSealList/CreateSealList/Forms/frmMain.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic.FileIO;

namespace CreateSealList.Forms
{
    public partial class frmMain : Form
    {
        public frmMain()
        {
            InitializeComponent();
        }

        #region フォームイベント

        /// <summary>
        /// フォームロード
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void frmMain_Load(object sender, EventArgs e)
        {
            this.Text = Application.ProductName + " Ver.";
            this.Text += Application.ProductVersion.ToString();

            Initialization();
        }

        /// <summary>
        /// フォームクローズ
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void frmMain_FormClosing(object sender, FormClosingEventArgs e)
        {
            XmlSettings.LoadFromXmlFile();
            XmlSettings.Instance.TextFile = this.txtCSV.Text;
            XmlSettings.Instance.OutFolder = this.txtOutFolder.Text;
            XmlSettings.SaveToXmlFile();
        }

        #endregion

        #region オブジェクトイベント

        /// <summary>
        /// テキストボックスドラッグイベント
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void txtFolder_DragEnter(object sender, DragEventArgs e)
        {
            if (e.Data.GetDataPresent(DataFormats.FileDrop))
            {
                e.Effect = DragDropEffects.Copy;
            }
            else
            {
                e.Effect = DragDropEffects.None;
            }
        }

        /// <summary>
        /// テキストボックスフォルダドロップイベント
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void txtFolder_DragDrop(object sender, DragEventArgs e)
        {
            string[] strFiles = (string[])e.Data.GetData(DataFormats.FileDrop, false);
            TextBox txtFile = (TextBox)sender;

            if (System.IO.File.Exists(strFiles[0]))
            {
                txtFile.Text = System.IO.Path.GetDirectoryName(strFiles[0]);
            }
            else
            {
                if (System.IO.Directory.Exists(strFiles[0]))
                {
                    txtFile.Text = strFiles[0];
                }
            }
        }

        /// <summary>
        /// テキストボックスフォルダドロップイベント
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void txtCSV_DragDrop(object sender, DragEventArgs e)
        {
            string[] strFiles = (string[])e.Data.GetData(DataFormats.FileDrop, false);
            TextBox txtFile = (TextBox)sender;

            if (System.IO.Path.GetExtension(strFiles[0]) == ".txt")
            {
                txtFile.Text = strFiles[0];
            }
            else
            {
                txtFile.Text = "";
            }
        }

        /// <summary>
        /// 開始ボタン押下時
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnRun_Click(object sender, EventArgs e)
        {
            if (System.IO.File.Exists(this.txtCSV.Text) == false)
            {
                MessageBox.Show("リストファイルを指定してください", "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return;
            }
            if (System.IO.Directory.Exists(this.txtOutFolder.Text) == false)
            {
                MessageBox.Show("存在する出力フォルダを指定してください", "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return;
            }

            if (MessageBox.Show("シールリストを作成します" + Environment.NewLine + "よろしいですか？", "確認", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.No)
            {
                return;
            }

            //出力ファイル名の命名
            string strDt = DateTime.Now.ToString("yyyyMMdd_HHmmss");
            string strOutFile = this.txtOutFolder.Text + "\\シール用_" + strDt + ".txt";
            var enc = System.Text.Encoding.GetEncoding("Shift-JIS");
            int iCount = 0;

            using (var sw = new System.IO.StreamWriter(strOutFile, false, enc))
            {

                using (var parser = new TextFieldParser(this.txtCSV.Text, enc))
                {
                    parser.Delimiters = new string[] { "\t" };

                    while (!parser.EndOfData)
                    {
                        var rows = parser.ReadFields();
                        //各要素を読み込む
                        var iSeq1 = int.Parse(rows[0]);
                        var iSeq2 = int.Parse(rows[3]);
                        var iTotal = int.Parse(rows[4]);
                        var iReturn = int.Parse(rows[5]);
                        //要素4の指定回数分繰り返す
                        for (int i = 0; i < iTotal; i++)
                        {
                            iCount += 1;
                            string strWriteLine = iSeq1 + "_" + iSeq2 + "," + iCount.ToString("0000") + ",";
                            if (iReturn != 0)
                            {
                                strWriteLine += "返却";
                                iReturn -= 1;
                            }
                            sw.WriteLine(strWriteLine);
                        }
                    }
                }
            }
            MessageBox.Show("シールリストの作成が完了しました", "確認", MessageBoxButtons.OK, MessageBoxIcon.Information);
        }

        #endregion

        #region プライベートメソッド

        /// <summary>
        /// 初期化処理
        /// </summary>
        private void Initialization()
        {
            XmlSettings.LoadFromXmlFile();
            this.txtCSV.Text = XmlSettings.Instance.TextFile;
            this.txtOutFolder.Text = XmlSettings.Instance.OutFolder;
        }
        #endregion
    }
}
